### ISSUE TITLE
Wavelink Performance

### ISSUE THREAD

**Issue by CoreParadox** [oddbear/Loupedeck.WaveLink.Plugin#2]
Now, I know that this is a beta release, so you may be very well aware of this issue as well.

I've noticed while this plugin is active, the WaveLink audio processor struggles a bit, with some cracking, skipping, and general audio distortion, it seems increasing the buffer size resolves this issue, but obviously not the ideal solution. 

There are some other small issues I've noticed, but again I'm assuming it's just due to the current in-dev status of the plugin.

Great work, I look forward to using it!

**Comment by oddbear** [comment]
Hi. 👋
Thanks for reporting.

The plugin is made with [ElgatoWaveLinkSDK](https://github.com/Professor-Melvin/ElgatoWaveLinkSDK) , so almost all kudos must go there. This library has just got a upgrade (18 hours ago), so I will test that one out to see if it helps. Might add a new pre release just with that build if you want to test to?

I have gotten some crackling, and also some crashes with the Wave Link software. Also that the plugin gets slow over time.
Not sure if this was only my machine, so good to hear that also other has this issue.

In a worst case scenario, I would need to create a library myself, but I really hope it does not come to that.

Is it ok for you to refer this issue (or for you to create another one) on the library repository, if the latest version does not solve the problems?

**Comment by CoreParadox** [comment]
> Is it ok for you to refer this issue (or for you to create another one) on the library repository, if the latest version does not solve the problems?

Absolutely, I do also have a streamdeck which has wavelink integration (but no dials) and I don't experience the issue there, although I suppose having both active could cause a conflict or something, and of course, the stream deck integration is from elgato themselves, so there may be some black box stuff happening there.

**Comment by oddbear** [comment]
Just updated the with the latest SDK.

If you want to try it out, you can try it from here:
https://github.com/oddbear/Loupedeck.WaveLink.Plugin/releases/tag/v0.5.0

I will however recommend to fully uninstall the old version first, as there might be conflicts with this one.
I can report back on this thread if I see any other issues with this.

**Comment by CoreParadox** [comment]
Seems good so far, I'll let you know as I use it for the next day or so!

**Comment by CoreParadox** [comment]
> Also that the plugin gets slow over time.
> Not sure if this was only my machine, so good to hear that also other has this issue.

I did have some crackling occur after a few hours of mostly continuous audio, I'll continue testing and maybe take a look through to see if I anything stands out

**Comment by oddbear** [comment]
New beta out 0.6.0.

I have tested with no connections to Wave Link, with the SDK Testing running, and with the plugin running.
Probably most of my issues was because of other things (cracking on high CPU usage).

This version is using "message queues" to see if the issue is Wave Link beeing blocked from updating the states. Things to look after is latency etc.

One thing to look after, is if this happens when using the Loupedeck knobs a lot. It might be that Wave Link just cannot handle such a load (as the Stream Deck is button based).
